core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/AsyncTaskProcessor.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+import cats.effect.{Concurrent, Timer}
+import cats.implicits._
+import fs2.Stream
+import fs2.concurrent.InspectableQueue
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.leonardo.AsyncTaskProcessor._
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+import scala.concurrent.duration._
+
+final class AsyncTaskProcessor[F[_]: Timer](config: AsyncTaskProcessor.Config, asyncTasks: InspectableQueue[F, Task[F]])(
+  implicit logger: Logger[F],
+  F: Concurrent[F],
+  metrics: OpenTelemetryMetrics[F]
+) {
+  def process: Stream[F, Unit] = {
+    val asyncTaskStream = asyncTasks
+      .dequeueChunk(config.maxConcurrentTasks)
+      .parEvalMapUnordered(config.maxConcurrentTasks)(task => handler(task))
```

Comment:
Curious about the bug you mentioned in standup.. is the fix essentially calling `handler` in `parEvalMapUnordered`? Maybe we can talk details @ walkthrough :)

Reply:
yea, this is the bug...It's definitely interesting to show...I added it on walkthrough agenda...I plan to run the unit tests and it'll become clear what happened